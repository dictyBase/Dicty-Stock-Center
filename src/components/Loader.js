import React from "react"
import Skeleton from "react-loading-skeleton"
import { Flex, Box } from "rebass"
import { Container } from "styles"

const Loader = () => {
  return (
    <Container>
      <Flex justify="center">
        <Box w={1}>
          <Skeleton count={10} />
          <br />
          <br />
          <Skeleton count={10} />
          <br />
          <br />
          <Skeleton count={10} />
        </Box>
      </Flex>
    </Container>
  )
}

export default Loader
