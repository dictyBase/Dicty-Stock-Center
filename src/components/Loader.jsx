import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Flex, Box } from 'rebass'
import styled from 'styled-components'
import FontAwesome from 'react-fontawesome'

const CenteredBox = styled(Box)`
    text-align: center;
`

export default class Loader extends Component {
    displayName = 'loading component'
    static propTypes = {
        title: PropTypes.string,
        message: PropTypes.string
    }
    render() {
        const { title, message } = this.props
        return (
            <Flex wrap justify="center">
                <CenteredBox>
                    <h1>{ title && title }</h1>
                    <FontAwesome name="spinner" pulse size="5x" fw />
                    <h4>{ message && message }</h4>
                </CenteredBox>
            </Flex>
        )
    }
}
