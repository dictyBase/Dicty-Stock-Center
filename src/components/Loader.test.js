import React from 'react'
import renderer from 'react-test-renderer'
import Loader from './Loader'

test('matching a snapshot of Loader', () => {
    const component = renderer.create(
    <Loader />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})
