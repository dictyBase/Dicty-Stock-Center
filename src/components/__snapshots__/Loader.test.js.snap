// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of Loader 1`] = `
<div
  className="sc-bxivhb bUZGRr sc-htpNat eDtRyC sc-bdVaJa iHZvIS"
  is={null}
>
  <div
    className="sc-bmyXtO cjEiu sc-ifAKCX cFlEyZ sc-bdVaJa iHZvIS"
    is={null}
  >
    <h1 />
    <span
      aria-hidden={true}
      className="fa fa-spinner fa-5x fa-pulse fa-fw"
    />
    <h4 />
  </div>
</div>
`;
